pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go

```diff
@@ -434,7 +440,7 @@ func deploymentHealthChecks(machineDeployment *capi.MachineDeployment, machinePo
 					Timeout: *machinePool.UnhealthyNodeTimeout,
 				},
 			},
-			MaxUnhealthy:       machinePool.MaxUnhealthy,
+			MaxUnhealthy:       &maxUnhealthy,
```

Comment:
I think that if the a value for `machinePool.MaxUnhealthy == nil` then `MaxUnhealthy` in the `MachineHealthCheck` should also be `nil`. In this implementation, if `machinePool.MaxUnhealthy == nil`, then `MaxUnhealthy` in the `MachineHealthCheck` is `0`.

Is that intended?

Reply:
good catch